### PR TITLE
IP/Top: Fix fallback IP address

### PR DESCRIPTION
**Pull request by sepalani**
I don't know if that will change anything but this PR should fix the IP fallback values.

Ready to be reviewed & merged.

### CODE REVIEW

**Review by BhaaLseN — APPROVED** [review]
I guess we can take that as-is, at least until we're also considering big endian systems /shrug
(untested though)

File: Source/Core/Core/IOS/Network/IP/Top.cpp [hunk 1]
```diff
@@ -258,8 +254,8 @@ static std::optional<DefaultInterface> GetSystemDefaultInterface()
 
 static DefaultInterface GetSystemDefaultInterfaceOrFallback()
 {
-  static constexpr DefaultInterface FALLBACK_VALUES{
-      inet_addr(10, 0, 1, 30), inet_addr(255, 255, 255, 0), inet_addr(10, 0, 255, 255)};
+  // Fallback values in little endian: {10.0.1.30, 255.255.255.0, 10.0.255.255}
+  static constexpr DefaultInterface FALLBACK_VALUES{0x0e01000a, 0x00ffffff, 0xffff000a};
```
Comment (leoetlino) [inline Source/Core/Core/IOS/Network/IP/Top.cpp:258]: Couldn't you just change our `inet_addr` to use a different endianness (or just add a call to Common::Swap32)? This is less readable.

**Comment by leoetlino** [comment]
This needs a rebase

File: Source/Core/Core/IOS/Network/IP/Top.cpp (hunk 1 above)
Comment (sepalani) [reply Source/Core/Core/IOS/Network/IP/Top.cpp]: I took a similar "swap" approach using the native `inet_addr` as it returns the value as expected by the DefaultInterface structure.

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Pull request merged**